### PR TITLE
simplified assertions and removed unthrown exceptions

### PR DESCRIPTION
**Pull request by StefanSpieker**
I simplified one if condition and simplified some test assertions and cleaned up some throws Exceptions and some minor improvements.

### Proposed changelog entries

* N/A

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]
looks good assuming the build passes

**Review by res0nance — APPROVED** [review]

**Review by jvz — APPROVED** [review]

File: core/src/main/java/hudson/model/AbstractProject.java [hunk 1]
```diff
@@ -520,9 +520,7 @@ private AbstractBuild getBuildForDeprecatedMethods() {
                     return b;
             }
         }
-        R lb = getLastBuild();
-        if(lb!=null)    return lb;
-        return null;
+        return getLastBuild();
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/AbstractProject.java]: Bonus points for marking the method as `CheckForNull`

**Review by oleg-nenashev — APPROVED** [review]
I plan to merge it tomorrow if no negative feedback

File: core/src/main/java/hudson/model/AbstractProject.java (hunk 1 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/model/AbstractProject.java]: Can be done in a follow-up

**Comment by oleg-nenashev** [comment]
@StefanSpieker  Is it a Hacktoberfest pull request just in case? Just to label it properly on our side. See [Hacktoberfest in Jenkins](https://jenkins.io/events/hacktoberfest). If you do not participate in it, please consider doing so. Your pull requests are much appreciated 👍

**Pull request merged**